### PR TITLE
feat: add preRemoveCallback

### LINKED ISSUE #72: Add pre-expiration callback for auditing and custom logic in connect-mssql-v2
**Is your feature request related to a problem? Please describe.**
Yes, the problem is related to the lack of a callback or hook that is invoked immediately before a session expires and is deleted from the database. Currently, the connect-mssql-v2 library provides the autoRemoveCallback option, which is triggered after the expired sessions are deleted. However, there is no way to execute custom logic or perform necessary actions, such as auditing, right before the session expiration process begins.

**Describe the solution you'd like**
The desired solution is to add support for a callback function that is invoked just before the session expiration process starts. This callback should be configurable through a new option, such as beforeExpireCallback, which accepts a function.

The destroyExpired method should be modified to execute the beforeExpireCallback (if provided) before running the delete query on the sessions table. The callback should receive relevant information, such as the session ID or session data, to allow for custom logic based on the expiring session.

**Describe alternatives you've considered**
Another alternative is to use a separate cron job or background process to query the sessions table, identify expiring sessions, and perform custom actions before deleting them. However, this approach can be complex to set up and maintain, and it may introduce race conditions if the session is modified or extended after being identified as expired.

**Additional context**
In my application, I implement auditing functionality to track important user actions, such as sign-ins and sign-outs, by writing relevant information to an audit table. However, when a session expires automatically due to inactivity or reaching the expiration time, this  library directly deletes the session data from the database without providing any hooks or callbacks that allow me to perform auditing or other necessary actions before or after the session expiration.

### PR DESCRIPTION
**Pull request by bradtaniguchi**
- provides `preRemoveCallback` option. If provided initially, is called when a session is expired, but before the actual "delete" code. If the function returns a promise, then the callback will wait for that promise to resolve before executing the "delete" code.

- also fix some minor typos

Close #72

### CODE REVIEW

File: .eslintcache
Comment (bradtaniguchi) [inline .eslintcache:1]: Let me add this to `.gitignore`, not sure why it appears now.

**Review by JLuboff — COMMENTED** [review]
Nice work and thank you for knocking this out! Thoughts on adding a test to the test suite?

**Comment by bradtaniguchi** [comment]
I think I can try to add in some. 

I'll have to review how to setup a full/accurate test environment on my WSL environment tho.

Might be doing the tests a little blind otherwise 😅

**Comment by bradtaniguchi** [comment]
I added a test case that follows a similar setup (but uses promises to keep things flat rather than nested callbacks), but I don't have a MSSQL database around to help run against so can't verify it.

**Comment by bradtaniguchi** [comment]
Ok I got an SQL instance from Azure (thx free tier), I confirmed the tests is working/going through from my end. I also ran a smoke check against the test (instead of checking the callback was called once, I changed it to 10) and confirmed the test fails. 

Also now that I have an SQL server to test against, I **should** be able to help debug/clean up this repo if needed, as I can finally test my changes against the test suite ^.^

**Comment by JLuboff** [comment]
Thanks for adding the test! Looks good to me (I ran locally). I will say...I don't know what I was thinking calling the variable `cbed` (callback ended??? I have no idea). After this PR gets merged I'll do a cleanup PR (and update deps). 
Glad you got SQL Server running :D I run a local docker build of SQL Server (express). Works well and only spin it up when I need to. I seem to recall pulling it down from Microsoft.

**Review by JLuboff — APPROVED** [review]
🔥 Thanks for the awesome contribution!

**Comment by bradtaniguchi** [comment]
> Thanks for adding the test! Looks good to me (I ran locally). I will say...I don't know what I was thinking calling the variable `cbed` (callback ended??? I have no idea). After this PR gets merged I'll do a cleanup PR (and update deps). Glad you got SQL Server running :D I run a local docker build of SQL Server (express). Works well and only spin it up when I need to. I seem to recall pulling it down from Microsoft.

Yea I have a new laptop that I can actually work on (previously was doing some convoluted stuff on a chromebook). Getting a docker setup running could also help test this stuff out if needed 👍🏼

**Comment by JLuboff** [comment]
@bradtaniguchi Not sure if you were waiting to merge this since it was failing the checks. I cleared that (it was complaining due to line length in the README which doesn't matter). However, I realized one thing. Need the version updated (minor?) in package.json and update the changelog. Can be a separate PR, this PR, or I can handle it. You decide. Will than release (I think I'll need to? I can't recall if we have you setup for permissions in npm or not).

**Comment by bradtaniguchi** [comment]
I've been meaning to get back to this, I can fix the build/bug issues in a week or so.

You can do the release, as I'm not sure how the NPM package setup is done off the top of my head.

**Comment by JLuboff** [comment]
Piggy-backed on your PR for the changelog and semver change so it can all be kept together.

**Pull request merged**